### ISSUE TITLE
SimpleMagnetometer sketch static reading 

### ISSUE THREAD

**Issue by Wh94** [arduino-libraries/Arduino_BMI270_BMM150#43]
As titled, I have no issues with the accelerometer and gyroscope sketches and those are seem to be working fine, however, for the Simplemagnetometer sketch, I am having a static measurement on all three axis as shown in the image below. Anyone has this issue? 

![image](https://github.com/user-attachments/assets/a3b13be4-3bb7-4b60-ae29-7c35ee38b981)

**Comment by CDBeats** [comment]
I'm having the same issue. Maybe there's something wrong with the hardware? I'm using an Arduino Nano 33 BLE Rev2, and I think after some update, the mag data just broke.

**Comment by Sidharth-NK** [comment]
> As titled, I have no issues with the accelerometer and gyroscope sketches and those are seem to be working fine, however, for the Simplemagnetometer sketch, I am having a static measurement on all three axis as shown in the image below. Anyone has this issue?
> 
> ![image](https://github.com/user-attachments/assets/a3b13be4-3bb7-4b60-ae29-7c35ee38b981)

I got this output

<img width="792" height="539" alt="Image" src="https://github.com/user-attachments/assets/5c59c567-c698-4361-824d-c7de2b9fe13e" />

**Comment by CDBeats** [comment]
> > As titled, I have no issues with the accelerometer and gyroscope sketches and those are seem to be working fine, however, for the Simplemagnetometer sketch, I am having a static measurement on all three axis as shown in the image below. Anyone has this issue?
> > ![image](https://github.com/user-attachments/assets/a3b13be4-3bb7-4b60-ae29-7c35ee38b981)
> 
> I got this output
> 
> <img alt="Image" width="792" height="539" src="https://private-user-images.githubusercontent.com/95956679/531872073-5c59c567-c698-4361-824d-c7de2b9fe13e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc2MjI4NDMsIm5iZiI6MTc2NzYyMjU0MywicGF0aCI6Ii85NTk1NjY3OS81MzE4NzIwNzMtNWM1OWM1NjctYzY5OC00MzYxLTgyNGQtYzdkZTJiOWZlMTNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA1VDE0MTU0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljNTJlMjgyM2RlMjNiYzIwNDE0Y2E4ZjZiNjYwMTQ4NzM0ODJlNzQ2M2Y2MWJiYzIxMmJkMmE3NDQ2NWQwZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Ol3R9oihodYmMrukrc3JnHC-hXQbU-6W4cRxjhd169Q">

Yeah I think it could be a hardware issue for me and op, maybe due to magnetic interference. Bright side is that magnetometers breakout boards are relatively cheap.
